From c0426cf37aad2cb56081206aa08652282cf6778f Mon Sep 17 00:00:00 2001
From: Benno Schulenberg <bensberg@justemail.net>
Date: Thu, 11 Aug 2016 12:37:11 +0200
Subject: [PATCH] input: try looking at the keyboard state only when compiled
 on Linux

---
 src/winio.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/winio.c b/src/winio.c
index bcfb8a2..5ab1e18 100644
--- a/src/winio.c
+++ b/src/winio.c
@@ -23,7 +23,9 @@
 #include "proto.h"
 #include "revision.h"
 
+#ifdef __linux__
 #include <sys/ioctl.h>
+#endif
 
 #include <stdio.h>
 #include <stdarg.h>
@@ -504,6 +506,7 @@ int parse_kbinput(WINDOW *win)
 	return sc_seq_or(do_next_block, 0);
 #endif
 
+#ifdef __linux__
     /* When not running under X, check for the bare arrow keys whether
      * the Ctrl key is being held together with them. */
     if (console && (retval == KEY_UP || retval == KEY_DOWN ||
@@ -521,6 +524,7 @@ int parse_kbinput(WINDOW *win)
 		return sc_seq_or(do_next_word_void, 0);
 	}
     }
+#endif
 
     switch (retval) {
 #ifdef KEY_SLEFT
-- 
2.9.2

